components/GifRenderer/common/src/desktopMain/kotlin/org/jetbrains/compose/gifrenderer/DesktopGifRenderer.kt

```diff
@@ -0,0 +1,50 @@
+package org.jetbrains.compose.gifrenderer
+
+import androidx.compose.animation.core.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.compose.ui.graphics.painter.BitmapPainter
+import androidx.compose.ui.graphics.painter.Painter
+import org.jetbrains.skija.Bitmap
+import org.jetbrains.skija.Codec
+import org.jetbrains.skija.Data
+import java.net.URL
+
+@Composable
+internal actual fun rememberGifRendererImpl(url: String): State<Painter> {
+    val codec = remember(url) {
+        println("Gif renderer for $url")
+        val bytes = URL(url).readBytes()
```

Comment:
Use IO dispatcher to fetching data from network.

Reply:
I think we can provide an universal function to load from any InputStream (see above). But we shouldn't decide for the users how they want to load an animation - synchronously or asynchronously.

So I suggest not to introduce the Composable API for loading from URL. In core Compose we also didn't introduce API for asynchronous loading (to be specific, we introduced it in the past, but removed [later](https://android-review.googlesource.com/c/platform/frameworks/support/+/1561297))